i18n-helpers/src/bin/mdbook-xgettext.rs

```diff
@@ -226,4 +422,168 @@ mod tests {
 
         Ok(())
     }
+
+    #[test]
+    fn test_create_catalog_nested_directories() -> anyhow::Result<()> {
+        let (ctx, _tmp) = create_render_context(&[
+            ("book.toml", "[book]"),
+            (
+                "src/SUMMARY.md",
+                "- [The Foo Chapter](foo.md)\n\
+                \t- [The Bar Section](foo/bar.md)\n\
+                \t\t- [The Baz Subsection](foo/bar/baz.md)",
+            ),
+            (
+                "src/foo.md",
+                "# How to Foo\n\
+                 \n\
+                 The first paragraph about Foo.\n",
+            ),
+            (
+                "src/foo/bar.md",
+                "# How to Bar\n\
+                 \n\
+                 The first paragraph about Bar.\n",
+            ),
+            (
+                "src/foo/bar/baz.md",
+                "# How to Baz\n\
+                 \n\
+                 The first paragraph about Baz.\n",
+            ),
+        ])?;
+
+        // Using a depth of 0 to include all messages in a single template file.
+        let catalogs = create_catalogs(&ctx, 0, "messages.pot")?;
+        let catalog = &catalogs[&default_template_file()];
+
+        for msg in catalog.messages() {
+            assert!(!msg.is_translated());
+        }
+
+        let expected_message_tuples = vec![
+            ("src/SUMMARY.md:1", "The Foo Chapter"),
+            ("src/SUMMARY.md:2", "The Bar Section"),
+            ("src/SUMMARY.md:3", "The Baz Subsection"),
+            ("src/foo.md:1", "How to Foo"),
+            ("src/foo.md:3", "The first paragraph about Foo."),
+            ("src/foo/bar.md:1", "How to Bar"),
+            ("src/foo/bar.md:3", "The first paragraph about Bar."),
+            ("src/foo/bar/baz.md:1", "How to Baz"),
+            ("src/foo/bar/baz.md:3", "The first paragraph about Baz."),
+        ];
+
+        let message_tuples = catalog
+            .messages()
+            .map(|msg| (msg.source(), msg.msgid()))
+            .collect::<Vec<(&str, &str)>>();
+
+        for message_tuple in expected_message_tuples {
+            assert!(message_tuples.contains(&message_tuple));
+        }
+
+        Ok(())
+    }
+
+    fn summary_file() -> PathBuf {
+        PathBuf::from("src/SUMMARY.md")
+    }
+    fn summary_content() -> Vec<&'static str> {
```

Comment:
Please drop this — it makes the tests hard to understand since it now requires knowledge of the `test_case` attribute below. I would prefer having two simple tests over this :slightly_smiling_face: